articles/understanding-dart-loops/index.md

```diff
@@ -0,0 +1,370 @@
+
+A Loop's definition is the code that executes multiple times while Iteration refers to the process in which the content executes once. One iteration refers to 1-time execution of a loop. It can undergo many iterations. They are a way of using something to an unknown or not innumerable amount of things. Some applications include adding sums of numbers, cycling through values, repeat functions, to mention a few.
+
+### To write Dart Programs
+To write Dart programs and implement the knowledge you need to have:
+
+- You can use the **[DartPad Online Editor](https://dartpad.dartlang.org/)** OR
+
+- You can use this method if you are familiar with [`Java`](https://en.wikipedia.org/wiki/Java_(programming_language)) or [`Kotlin`](https://en.wikipedia.org/wiki/Kotlin_(programming_language)) , **[Install Dart SDK ](https://dart.dev/get-dart)**, **[Install IntelliJ IDEA ](https://www.jetbrains.com/idea/)** and integrate the Dart plugin to IntelliJ IDEA IDE.
+
+### Table of Contents
+- [General Loop Structure](#general-loop-structure)
+- [Loops in Dart](#loops-in-dart)
+   - [For Loop](#for-loop)
+   - [While Loop](#while-loop)
+   - [Do-While Loop](#do-while-loop)
+- [Other Key Concepts](#other-key-concepts)
+   - [Break Keyword](#break-keyword) 
+   - [Continue Keyword](#continue-keyword) 
+- [Applications of Loops](#applications-of-loops)
+- [Further Reading](#further-reading)
+- [Conclusion ](#conclusion)
+
+### General Loop Structure
+There are two types of loops:
+1. **Definite Loops**
+These are loops that we know the number of times we want to execute the code.***For example***, The `For Loop`, later on through this article, we shall come to understand that it has a definite number of iterations known to the programmer.
+
+2. **Indefinite Loops**
+These refer to loops we do not know the number of times we want to execute the code. ***For example***, The `While Loop` and The `Do-While Loop`, later on, we shall come to understand what it means by the iterations not known to the programmer.
+
+All the loop types mentioned above have the following characteristics:
+
+- *Counter Variable* - It is also known as the Initializer. It is the starting point where it executes.
+
+- *Increment or Decrement Counter Variable*  - This refers to the number in which counter variable increases or decreases it meets the condition.
+
+- *Condition Check* - Every loop will have a condition which will be checked on each Iteration. If the condition is evaluated to true, the next iteration will get executed.
+
+  There are two types of loops in condition check:
+
+  1. Entry controlled loops - These are loops in which the condition is checked first then the body of the loop executes.
+
+  2. Exit controlled loops - These are loops in which the body of the loop executes then the condition check afterwards.
```

Comment:
Rewrite as:

These are loops in which an Iteration is executed and then the condition is checked afterward.